### PR TITLE
Update from deprecated aws-sdk bindings

### LINKED ISSUE #629: Update to v3 of AWS SDK
Executing the S3 code gives this warning:

> NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

There is a TODO in the source code related to updating our S3 bindings for the next version of the AWS SDK:

https://github.com/purescript/registry-dev/blob/dc2e7886caf13da3d7b2c2ed32950ff8a731944b/foreign/src/Foreign/S3.js#L12-L13

### PR DESCRIPTION
**Pull request by thomashoneyman**
Fixes #629. I manually verified that the inputs (params we pass to the s3 object) and outputs (data we receive) are unchanged in the new SDK, but I haven't actually run this code against our live bucket because the giant legacy importer run is going.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -480,7 +480,7 @@
                         {
                           request = {
                             method = "PUT";
-                            url = "/effect/4.0.0.tar.gz";
+                            url = "/effect/4.0.0.tar.gz?x-id=PutObject";
```
Comment (thomashoneyman) [inline flake.nix:483]: This is the only visible change to the requests based on our integration test and I _think_ it's harmless, but without verifying against the actual bucket I can't be completely sure.

Comment (f-f) [reply flake.nix:483]: If it's generated by the new sdk then it's probably the right thing 😄

**Review by f-f — APPROVED** [review]
Ah, this looks way more straightforward than the last time I tried it 😅

**Comment by f-f** [comment]
Now that you're also online we can merge this and eventually revert it real quick if it's broken 😄

**Comment by thomashoneyman** [comment]
I was actually putting together a little script to upload, download, and delete an object in storage to be sure before I merge :) Let me give that a quick try and if it works I'll merge! I'll need about 1.5hrs before I can do it.

**Comment by f-f** [comment]
Ah that's lovely, thank you 🙂

**Comment by thomashoneyman** [comment]
Ran this script to test things out, with a tarball at `delete-me-1.0.0` locally:

```purs
ain :: Effect Unit
main = launchAff_ do
  -- Environment
  _ <- Env.loadEnvFile ".env"
  resources <- Env.lookupResourceEnv
  s3 <- lift2 { key: _, secret: _ } (Env.lookupRequired Env.spacesKey) (Env.lookupRequired Env.spacesSecret)

  let
    cache = Path.concat [ scratchDir, ".cache" ]

    work = do
      Storage.upload (unsafePackageName "delete-me") (unsafeVersion "1.0.0") (Path.concat [ scratchDir, "delete-me-1.0.0" ])
      versions <- Storage.query (unsafePackageName "delete-me")
      Console.log $ "Versions (uploaded): " <> String.joinWith ", " (map Version.print (Set.toUnfoldable versions))
      Storage.download (unsafePackageName "delete-me") (unsafeVersion "1.0.0") (Path.concat [ scratchDir, "delete-me-2.0.0" ])
      Storage.delete (unsafePackageName "delete-me") (unsafeVersion "1.0.0")
      versions2 <- Storage.query (unsafePackageName "delete-me")
      Console.log $ "Versions (deleted): " <> String.joinWith ", " (map Version.print (Set.toUnfoldable versions2))

  work
    # Storage.interpret (Storage.handleS3 { s3, cache })
    # Log.interpret (Log.handleTerminal Verbose)
    # Except.catch (\e -> Run.liftEffect (Console.log e *> Process.exit 1))
    # Env.runResourceEnv resources
    # Run.runBaseAff'
```

This works just fine with the v2 of aws-sdk:

```console
Read file for delete-me@1.0.0, now uploading to delete-me/1.0.0.tar.gz...
Connecting to the bucket purescript-registry at space https://ams3.digitaloceanspaces.com with public key YMQJ24GQQ2ZOGFXZWVQB
Connected to S3!
Uploading release to the bucket at path delete-me/1.0.0.tar.gz
Uploaded delete-me@1.0.0 to the bucket at path delete-me/1.0.0.tar.gz
Connecting to the bucket purescript-registry at space https://ams3.digitaloceanspaces.com with public key YMQJ24GQQ2ZOGFXZWVQB
Connected to S3!
Versions (uploaded): 1.0.0
Deleting delete-me@1.0.0
Connecting to the bucket purescript-registry at space https://ams3.digitaloceanspaces.com with public key YMQJ24GQQ2ZOGFXZWVQB
Connected to S3!
Deleting release from the bucket at path delete-me/1.0.0.tar.gz
Deleted release of delete-me@1.0.0 from S3 at the path delete-me/1.0.0.tar.gz
Connecting to the bucket purescript-registry at space https://ams3.digitaloceanspaces.com with public key YMQJ24GQQ2ZOGFXZWVQB
Connected to S3!
Versions (deleted):
```

However, it turns out that the `Contents` key is now optional in the response; with that fixed this PR works as well.

**Pull request merged**